### PR TITLE
[SPARK-27404][CORE][SQL][STREAMING][YARN] Fix build warnings for 3.0: postfixOps edition

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Fix build warnings -- see some details below.

But mostly, remove use of postfix syntax where it causes warnings without the `scala.language.postfixOps` import. This is mostly in expressions like "120000 milliseconds". Which, I'd like to simplify to things like "2.minutes" anyway.

## How was this patch tested?

Existing tests.

### CODE REVIEW

File: common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RetryingBlockFetcherSuite.java [hunk 1]
```diff
@@ -46,9 +46,9 @@
  */
 public class RetryingBlockFetcherSuite {
 
-  ManagedBuffer block0 = new NioManagedBuffer(ByteBuffer.wrap(new byte[13]));
-  ManagedBuffer block1 = new NioManagedBuffer(ByteBuffer.wrap(new byte[7]));
-  ManagedBuffer block2 = new NioManagedBuffer(ByteBuffer.wrap(new byte[19]));
+  private final ManagedBuffer block0 = new NioManagedBuffer(ByteBuffer.wrap(new byte[13]));
```
Comment (srowen) [inline common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RetryingBlockFetcherSuite.java]: OK, this is stuff I just cleaned up while touching this file.

File: common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RetryingBlockFetcherSuite.java [hunk 2]
```diff
@@ -291,7 +291,7 @@ private static void performInteractions(List<? extends Map<String, Object>> inte
     }
 
     assertNotNull(stub);
-    stub.when(fetchStarter).createAndStart(any(), anyObject());
+    stub.when(fetchStarter).createAndStart(any(), any());
```
Comment (srowen) [inline common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RetryingBlockFetcherSuite.java]: This was an avoidable deprecated method

File: core/src/main/scala/org/apache/spark/BarrierTaskContext.scala [hunk 3]
```diff
@@ -122,7 +121,7 @@ class BarrierTaskContext private[spark] (
           barrierEpoch),
         // Set a fixed timeout for RPC here, so users shall get a SparkException thrown by
         // BarrierCoordinator on timeout, instead of RPCTimeoutException from the RPC framework.
-        timeout = new RpcTimeout(31536000 /* = 3600 * 24 * 365 */ seconds, "barrierTimeout"))
+        timeout = new RpcTimeout(365.days, "barrierTimeout"))
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/BarrierTaskContext.scala]: A good example of simplifying expressions while also removing postfixOps

File: mllib/src/test/scala/org/apache/spark/ml/MLEventsSuite.scala [hunk 4]
```diff
@@ -245,18 +244,18 @@ class MLEventsSuite
       val pipelineReader = Pipeline.read
       assert(events.isEmpty)
       pipelineReader.load(path)
-      eventually(timeout(10 seconds), interval(1 second)) {
+      eventually(timeout(10.seconds), interval(1.second)) {
         events.foreach {
-          case e: LoadInstanceStart[PipelineStage]
+          case e: LoadInstanceStart[_]
```
Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/ml/MLEventsSuite.scala]: Can't match generic types - causes a warning

File: sql/core/src/test/java/test/org/apache/spark/sql/sources/v2/JavaSimpleBatchTable.java [hunk 5]
```diff
@@ -52,53 +48,3 @@ public String name() {
   }
 }
 
-abstract class JavaSimpleScanBuilder implements ScanBuilder, Scan, Batch {
```
Comment (srowen) [inline sql/core/src/test/java/test/org/apache/spark/sql/sources/v2/JavaSimpleBatchTable.java]: These were moved to top-level classes. Multiple top-level classes in Java are mostly a no-no, and actually generate a warning if you try to use them in another file.

**Review by felixcheung — APPROVED** [review]
LGTM. very long diff.

**Comment by srowen** [comment]
Wait until you see what has to happen to get rid of (mis)use of `scala.language.existentials`, which has been hiding some wrong usage of generics

File: core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala [hunk 6]
```diff
@@ -658,7 +657,7 @@ abstract class RpcEnvSuite extends SparkFunSuite with BeforeAndAfterAll {
       })
       val rpcEndpointRef = remoteEnv.setupEndpointRef(localEnv.address, "send-authentication")
       rpcEndpointRef.send("hello")
-      eventually(timeout(5 seconds), interval(10 millis)) {
+      eventually(timeout(5.seconds), interval(10.millis)) {
```
Comment (attilapiros) [inline core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala]: `10.millis` => `10.milliseconds`

There are a few other places where this pattern is repeated:
```
(pr/24314) $ git diff 017919b636  | grep "^\+.*millis[^e]"
+    eventually(timeout(1.minute), interval(10.millis)) {
+        rpcEndpointRef.askSync[String]("hello", new RpcTimeout(1.millis, shortProp))
+      eventually(timeout(5.seconds), interval(10.millis)) {
+    eventually(timeout(1.minute), interval(200.millis)) {
+      Await.ready(f, 50.millis)
+      eventually(timeout(20.seconds), interval(batchDuration.milliseconds.millis)) {
+      eventually(timeout(10.seconds), interval(10.millis)) {
```

Comment (attilapiros) [reply core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala]: Of course this is just a Nit as both correct (but from your previous changes I have the feeling milliseconds is prefered).

**Review by attilapiros — APPROVED** [review]
I have run some completeness grep (for finding second/milli/minute without dot prefix outside of string literals and comments) and this PR seams to me complete.

Regarding removing the unnecessary `case` after `map` and `foreach` expressions there are still cases Like:
https://github.com/apache/spark/blob/215609def22da14c464b37374ceae4f53a39a145/core/src/test/scala/org/apache/spark/ShuffleSuite.scala#L402

But as I got it that is not the scope of this change.

So LGTM.

**Comment by srowen** [comment]
Thanks @attilapiros yeah I fixed up more of those instances. Yes I have taken the liberty of also 'fixing' some minor things like unnecessary case statements etc in nearby code but tried to keep it mostly to build warnings and postfixOps here.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**